/Users/biancaganescu/School/transcendence-transformer/transcendenceGPT/train.py:138: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  get_batch.train_data = torch.load(os.path.join(data_dir, 'train.pt'))
tensor([[ 0,  6, 10, 11, 16,  2,  6, 10, 11, 16,  2,  6, 10, 13, 16,  2,  4,  1,
          0,  5,  9, 12, 15,  2,  6,  8, 13, 16,  2,  7, 10, 11, 14,  2,  3,  1,
          0,  7, 10, 13, 15,  2,  6, 10, 12, 14,  2,  5,  8, 13, 15,  2,  4,  1,
          0,  7,  9, 12, 15,  2,  6,  8, 12, 14,  2,  6,  8, 13, 14,  2,  4,  1,
          0,  7,  9, 11, 15,  2,  5,  8, 13, 15,  2,  5, 10, 12, 16,  2,  4,  1,
          0,  5,  9, 11, 16,  2,  6,  8, 13, 16,  2,  7,  8, 12, 16,  2,  4,  1,
          0,  5,  9, 13, 16,  2,  6,  8, 12, 15,  2,  6,  8, 13, 14,  2,  4,  1,
          0,  6,  9, 12, 15,  2,  7,  8, 13, 16,  2,  5, 10, 11, 15,  2,  3,  1,
          0,  7,  8, 13, 15,  2,  7, 10, 12, 15,  2,  5,  9, 12, 16,  2,  4,  1,
          0,  6, 10, 11, 16,  2,  7, 10, 13, 14,  2,  7, 10, 13, 15,  2,  4,  1,
          0,  7,  8, 11, 16,  2,  6,  9, 11, 14,  2,  5,  9, 11, 14,  2,  4,  1,
          0,  5, 10, 13, 15,  2,  7,  9, 11, 15,  2,  6,  8, 12, 15,  2,  3,  1,
          0,  6,  9, 13, 16,  2,  5,  9, 12, 14,  2,  7,  9, 11, 15,  2,  3,  1,
          0,  5,  9, 13, 14,  2,  7,  8, 13, 16,  2,  6, 10, 11, 15,  2,  4,  1,
          0,  7,  8, 13, 15,  2,  5,  9, 12, 15,  2,  6,  8, 12, 15,  2,  4,  1,
          0,  7, 10, 13, 14,  2,  5,  8, 11, 15,  2,  6,  9, 12, 16,  2,  3,  1,
          0,  5, 10, 13, 14,  2,  7, 10, 11, 15,  2,  6,  9, 12, 14,  2,  4,  1,
          0,  6, 10, 12, 14,  2,  7,  8, 13, 16,  2,  5,  9, 11, 15,  2,  3,  1,
          0,  6,  9, 13, 16,  2,  5,  8, 12, 14,  2,  7,  9, 11, 15,  2,  4,  1,
          0,  5,  9, 12, 16,  2,  6, 10, 11, 16,  2,  7,  8, 13, 16,  2,  3,  1,
          0,  7,  8, 13, 15,  2,  7,  9, 13, 15,  2,  6, 10, 13, 15,  2,  4,  1,
          0,  5,  9, 12, 14,  2,  6,  8, 11, 16,  2,  5, 10, 13, 15,  2,  4,  1,
          0,  5,  9, 11, 14,  2,  6, 10, 11, 15,  2,  7,  8, 13, 16,  2,  4,  1,
          0,  6,  8, 13, 15,  2,  6, 10, 11, 16,  2,  5,  9, 13, 16,  2,  4,  1,
          0,  5,  9, 13, 14,  2,  6, 10, 11, 15,  2,  6,  8, 13, 16,  2,  4,  1,
          0,  7, 10, 11, 14,  2,  7,  8, 12, 15,  2,  5, 10, 11, 15,  2,  4,  1,
          0,  7,  9, 12, 14,  2,  6,  9, 11, 14,  2,  7,  8, 13, 15,  2,  4,  1,
          0,  5,  9, 13, 16,  2,  5,  9, 12, 15,  2,  6,  9, 11, 16,  2,  4,  1,
          0,  5,  8, 12, 15,  2,  7,  9, 11, 14,  2,  6,  8, 13, 16,  2,  4,  1,
          0,  6,  8, 11, 15,  2,  6, 10, 12, 15,  2,  5,  8, 13, 16,  2,  4,  1]],
       dtype=torch.int32)
Traceback (most recent call last):
  File "/Users/biancaganescu/School/transcendence-transformer/transcendenceGPT/train.py", line 310, in <module>
    losses = estimate_loss()
  File "/Users/biancaganescu/anaconda3/envs/transcendence/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/Users/biancaganescu/School/transcendence-transformer/transcendenceGPT/train.py", line 270, in estimate_loss
    logits, loss = model(X, Y)
  File "/Users/biancaganescu/anaconda3/envs/transcendence/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/biancaganescu/anaconda3/envs/transcendence/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/biancaganescu/School/transcendence-transformer/transcendenceGPT/model.py", line 189, in forward
    loss = F.cross_entropy(logits.view(-1, logits.size(-1)), targets.view(-1), ignore_index=-1)
  File "/Users/biancaganescu/anaconda3/envs/transcendence/lib/python3.10/site-packages/torch/nn/functional.py", line 3479, in cross_entropy
    return torch._C._nn.cross_entropy_loss(
ValueError: Expected input batch_size (540) to match target batch_size (539).
